dart/dynamics/Joint.h

```diff
@@ -119,6 +126,14 @@ class Joint
   /// Get number of generalized coordinates
   DEPRECATED(4.1) virtual size_t getDof() const = 0;
 
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual DegreeOfFreedom* getDof(size_t _index) = 0;
+
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual const DegreeOfFreedom* getDof(size_t _index) const = 0;
```

Comment:
It makes sense. Let's not to add the function to Joint.

I'm just wondering that the Joint might be responsible for DOF name management since DOF is belongs to Joint. It would be okay that the DOF name is unique in the Joint. The name uniqueness in Skeleton is guaranteed by **full name**. The full name would be like "[SkeletonName].[JointName].[DofName]". But this idea is not finalized yet.